PlusServerLauncher/PlusServerLauncherMainWindow.cxx

```diff
@@ -890,3 +963,167 @@ void PlusServerLauncherMainWindow::OnTimerTimeout()
+  vtkSmartPointer<vtkXMLDataElement> startServerElement = vtkSmartPointer<vtkXMLDataElement>::New();
+  startServerElement->SetName("ServerStarted");
+  startServerElement->SetAttribute("LogLevel", logLevelString.c_str());
+  startServerElement->SetAttribute("ConfigFileName", filename.c_str());
+
+  vtkSmartPointer<vtkXMLDataElement> configFileElement = NULL;
+  configFileElement = vtkSmartPointer<vtkXMLDataElement>::Take(vtkXMLUtilities::ReadElementFromFile(configFilePath.c_str()));
+  if (configFileElement)
+  {
+    std::string ports = this->GetServersFromConfigFile(filename);
+    startServerElement->SetAttribute("Servers", ports.c_str());
+    startServerElement->AddNestedElement(configFileElement);
+  }
+  commandElement->AddNestedElement(startServerElement);
+
+  igtl::MessageBase::MetaDataMap map;
+  map["LogLevel"] = std::pair<IANA_ENCODING_TYPE, std::string>(IANA_TYPE_US_ASCII, startServerElement->GetAttribute("LogLevel"));
+  map["ConfigFileName"] = std::pair<IANA_ENCODING_TYPE, std::string>(IANA_TYPE_US_ASCII, startServerElement->GetAttribute("ConfigFileName"));
+  if (startServerElement->GetAttribute("Servers"))
+  {
+    map["Servers"] = std::pair<IANA_ENCODING_TYPE, std::string>(IANA_TYPE_US_ASCII, startServerElement->GetAttribute("Servers"));
+  }
+
+  std::stringstream commandStream;
+  vtkXMLUtilities::FlattenElement(commandElement, commandStream);
+  this->SendCommand(PlusServerLauncherMainWindow::PLUS_SERVER_LAUNCHER_REMOTE_DEVICE_ID, "ServerStarted", commandStream.str(), map);
+
+}
+
+//---------------------------------------------------------------------------
+void PlusServerLauncherMainWindow::SendServerStoppedCommand(std::string configFilePath)
+{
+  LOG_TRACE("Sending server stopped command");
+
+  vtkSmartPointer<vtkXMLDataElement> commandElement = vtkSmartPointer<vtkXMLDataElement>::New();
+  commandElement->SetName("Command");
+
+  igtl::MessageBase::MetaDataMap map;
+  map["ConfigFileName"] = std::pair<IANA_ENCODING_TYPE, std::string>(IANA_TYPE_US_ASCII, configFilePath);
+
+  std::stringstream commandStream;
+  vtkXMLUtilities::FlattenElement(commandElement, commandStream);
+  this->SendCommand(PlusServerLauncherMainWindow::PLUS_SERVER_LAUNCHER_REMOTE_DEVICE_ID, "ServerStopped", commandStream.str(), map);
+}
+
+//---------------------------------------------------------------------------
+void PlusServerLauncherMainWindow::OnLogEvent(vtkObject* caller, unsigned long eventId, void* clientData, void* callData)
+{
+  PlusServerLauncherMainWindow* self = reinterpret_cast<PlusServerLauncherMainWindow*>(clientData);
+
+  if (!self->m_RemoteControlServerConnector || !self->m_RemoteControlServerConnector->IsConnected())
+  {
+    return;
+  }
+
+  QString logMessage = QString();
+  const char* logMessageChar = static_cast<char*>(callData);
+  logMessage = logMessage.fromLatin1(logMessageChar);
+  if (logMessage.size() == 1) // If the length of the string is 1, then it may be a wchar_t*

```

Comment:
This is the current solution that I came up with to handle char* and wchar_t*.
It seems to work but I feel like there should be a better solution.

I've also found that wstring aren't displayed correctly in the QPlusStatusIcon.
I can apply whatever solution that is found there too.

Reply:
I've changed vtkPlusLogger to use a new vtkPlusLogger::MessageLogged and vtkPlusLogger::WideMessageLogged events. You'll have to have two observers/functions, but it's a better long term solution I think.